communicator/ssh/provisioner.go

```diff
@@ -106,8 +113,15 @@ func parseConnectionInfo(s *terraform.InstanceState) (*connectionInfo, error) {
 	if connInfo.Port == 0 {
 		connInfo.Port = DefaultPort
 	}
-	if connInfo.ScriptPath == "" {
-		connInfo.ScriptPath = DefaultScriptPath
+	// Set default targetPlatform to unix
+	if connInfo.TargetPlatform == "" {
+		connInfo.TargetPlatform = DefaultTargetPlatform
+	}
```

Comment:
It would be nice to validate the value of `TargetPlatform` here to make sure that it is one of our accepted values. If not, we can return an error to make sure that users don't type e.g. `"Windows"` and wonder why their connection fails.